spirv_glsl.cpp

```diff
@@ -6372,13 +6511,201 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
+		string expr;
+		expr = join("readFirstInvocationARB(" + to_expression(ops[2]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_ARB_shader_ballot");
+		break;
+	}
+
+	case OpSubgroupReadInvocationKHR:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("readInvocationARB(" + to_expression(ops[2]) + ", " +
+			to_expression(ops[3]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_ARB_shader_ballot");
+		break;
+	}
+
+	case OpSubgroupAllKHR:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("allInvocationsARB(" + to_expression(ops[2]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_ARB_shader_group_vote");
+		break;
+	}
+
+	case OpSubgroupAnyKHR:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("anyInvocationARB(" + to_expression(ops[2]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_ARB_shader_group_vote");
+		break;
+	}
+
+	case OpSubgroupAllEqualKHR:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("allInvocationsEqualARB(" + to_expression(ops[2]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_ARB_shader_group_vote");
+		break;
+	}
+
+	case OpGroupIAddNonUniformAMD:
+	case OpGroupFAddNonUniformAMD:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("addInvocationsNonUniformAMD(" + to_expression(ops[4]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_AMD_shader_ballot");
+		break;
+	}
+
+	case OpGroupFMinNonUniformAMD:
+	case OpGroupUMinNonUniformAMD:
+	case OpGroupSMinNonUniformAMD:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("minInvocationsNonUniformAMD(" + to_expression(ops[4]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_AMD_shader_ballot");
+		break;
+	}
+
+	case OpGroupFMaxNonUniformAMD:
+	case OpGroupUMaxNonUniformAMD:
+	case OpGroupSMaxNonUniformAMD:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+		expr = join("maxInvocationsNonUniformAMD(" + to_expression(ops[4]) + ")");
+		emit_op(result_type, id, expr, true);
+
+		require_extension("GL_AMD_shader_ballot");
+		break;
+	}
+
+	case OpFragmentMaskFetchAMD:
+	{
+		auto &type = expression_type(ops[2]);
+		uint32_t result_type = ops[0];
+		uint32_t id = ops[1];
+		string expr;
+
+		if (type.image.dim == spv::DimSubpassData)
+		{
+			expr = join("fragmentMaskFetchAMD(" + to_expression(ops[2]) + ")");
+		}
+		else {
```

Comment:
Bracing. You might want to give the clang-format a shot (with ./format_all.sh), but remember to commit first just in case :')